streamparse/dsl/bolt.py

```diff
@@ -3,69 +3,33 @@
-            raise TypeError("Invalid bolt: {}".format(component_cls))
-
-        super(BoltSpecification, self).__init__(component_cls, **kwargs)
-
-        # Ensure that the source is always a list.
-        if isinstance(source, string_types):
-            sources = [source]
-        elif isinstance(source, Iterable):
-            sources = list(source)
-        elif source:
-            sources = [source]
-        else:
-            raise TopologyError('{} component "{}" requires at least one source'
-                                .format(self.component_cls.__name__, self.name))
-
-        if not Grouping.valid(group_on):
-            raise TypeError('Invalid group_on given: {}'.format(group_on))
-
-        self.sources = sources
-        self.group_on = group_on
-
-    def resolve_dependencies(self, specifications):
-        """Modifies the bolt's sources to be references to other Specification
-        objects. Also ensures that group_on fields are provided by the
-        source Specifications.
-
-        :param specifications: all of the specification objects for this
-                               topology.
-        :type specifications: dict
-        """
-        for i, source in enumerate(self.sources):
-
-            # Resolve sources that are not a specification object.
-            if not isinstance(source, Specification):
-                if source in specifications:
-                    self.sources[i] = specifications[source]
-                else:
-                    raise TopologyError(
-                        'Missing source named {} from {}'.format(source, self))
-
-                # Ensure that source always references a specification object.
-                source = self.sources[i]
-
-            # Ensure that group on field names exist in the source.
-            if isinstance(self.group_on, list):
-                outputs = source.component_cls.streams
-
-                for group in self.group_on:
-                    if group not in outputs:
-                        raise TopologyError(
-                            'Field {} does not exist in source {} as an '
-                            'output'.format(group, outputs))
+from __future__ import absolute_import
+
+from .component import JavaComponentSpec, ShellComponentSpec
+
+
+class ShellBoltSpec(ShellComponentSpec):
+    def __init__(self, component_cls, name=None, command=None, script=None,
+                 inputs=None, par=1, config=None, outputs=None):
+        super(ShellBoltSpec, self).__init__(component_cls,
+                                            name=name,
+                                            inputs=inputs,
+                                            par=par,
+                                            config=config,
+                                            outputs=outputs,
+                                            command=command,
+                                            script=script)
```

Comment:
I think we're pretty close to always following the "visual" style where you align with the opening paren.  The only variance from that is when doing it that way will still put you over 80 characters.